extensions/amp-script/0.1/amp-script.js

```diff
@@ -288,21 +288,26 @@ export class AmpScript extends AMP.BaseElement {
     };
 
     // Create worker and hydrate.
-    WorkerDOM.upgrade(
-      container || this.element,
-      workerAndAuthorScripts,
-      config
-    ).then((workerDom) => {
-      this.workerDom_ = workerDom;
-      this.initialize_.resolve();
-      this.workerDom_.onerror = (errorEvent) => {
-        errorEvent.preventDefault();
-        user().error(
-          TAG,
-          `${errorEvent.message}\n    at (${errorEvent.filename}:${errorEvent.lineno})`
-        );
-      };
-    });
+    WorkerDOM.upgrade(container || this.element, workerAndAuthorScripts, config)
+      .then((workerDom) => {
+        this.workerDom_ = workerDom;
+        this.initialize_.resolve();
+        this.workerDom_.onerror = (errorEvent) => {
+          errorEvent.preventDefault();
+          user().error(
+            TAG,
+            `${errorEvent.message}\n    at (${errorEvent.filename}:${errorEvent.lineno})`
+          );
+        };
+      })
+      .catch((err) => {
+        // Exclude all errors created via workerAndAuthorScripts rejection.
+        if (err.message.indexOf('amp-script') > -1) {
```

Comment:
This seems brittle. Is there not a better way to distinguish the fetch failure from the script failure?

Reply:
If you read the description:

> The cleanest way of having the same effect would be to have the entire upgrade path within a .then() callback of the scripts, although that would then delay the start of the upgrade until after they've resolved (not that I think worker-dom actually does much work before they do, but it is what caused my hesitation).

Reply:
ah. just thought of something. hacky but not brittle